integration/tests/runtime-workflow-mongo.ts

```diff
@@ -0,0 +1,437 @@
+/* eslint-disable no-null/no-null */
+/* eslint-disable no-shadow */
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable import/no-extraneous-dependencies */
+import { mkdirSync, readFileSync, rmdirSync } from 'fs';
+import * as path from 'path';
+import { ApolloServer } from "apollo-server";
+import { createTestClient, ApolloServerTestClient } from 'apollo-server-testing';
+import { loadConfig } from 'graphql-config';
+import { loadDocuments } from '@graphql-toolkit/core';
+import { GraphQLFileLoader } from '@graphql-toolkit/graphql-file-loader';
+import { GraphbackGenerator, buildGraphbackAPI, GraphbackAPI } from "graphback";
+import { DocumentNode } from 'graphql';
+import { createMongoDbProvider } from "../../packages/graphback-runtime-mongodb"
+import { MongoClient, Db } from 'mongodb';
+import { SchemaCRUDPlugin } from '../../packages/graphback-codegen-schema';
+import { ClientCRUDPlugin } from '../../packages/graphback-codegen-client';
+
+/** global config */
+let db: Db;
+let mongoClient: MongoClient;
+let server: ApolloServer;
+let client: ApolloServerTestClient;
+let graphbackApi: GraphbackAPI;
+
+let documents: DocumentNode;
+
+let notesId = [];
+let commentId = [];
+let metadataId = [];
+
+const modelText = readFileSync("./mock.graphql").toString();
+
+beforeAll(async () => {
+  try {
+    const { graphbackConfig } = await getConfig();
+    mkdirSync("./output-mongo");
+    mkdirSync("./output-mongo/client")
+    const generator = new GraphbackGenerator(modelText, graphbackConfig);
+    generator.generateSourceCode();
+
+    mongoClient = new MongoClient('mongodb://mongodb:mongo@localhost:27017/users?authSource=admin', { useUnifiedTopology: true });
+    await mongoClient.connect();
+    db = mongoClient.db('users');
+    graphbackApi = buildGraphbackAPI(modelText, {
+      dataProviderCreator: createMongoDbProvider(db),
+      plugins: [
+        new SchemaCRUDPlugin({outputPath: "./output-mongo/schema/schema.graphql"}),
+        new ClientCRUDPlugin({format: 'graphql', outputFile: './output-mongo/client/graphback.graphql'})
+      ]
```

Comment:
Later on we could have the datasync plugin and add related tests here
/cc @ssd71

Reply:
Sounds good